### ISSUE TITLE
AoC Day 14: Calculate size programatically

### ISSUE THREAD

**Issue by salameme** [fasterthanlime/feedback#148]
**Where's the problem at?**
https://fasterthanli.me/series/advent-of-code-2022/part-14

**No, seriously, where is it?**
The floor is supposed to be infinite, but the input isn't, so, using the time-travel powers of editing this article, I'm going to decide on a reasonable size for the floor:

**Describe the issue and a fix if you know it**
You can actually programatically determine the dimensions of the grid:

Height: `height = ycoords.max()+2`
Width: `width = 500-height..=500+height`

This works because for the source to be covered you need sand or rocks left and right of it, drawing a diagonal from the floor to the source makes it a triangle, where every point on the edge is the same Manhattan distance from the point on the floor directly under the source.

In other words, use the formula above, and bar any off by one errors you should have a big enough grid.

**Comment by fasterthanlime** [comment]
I added a note:

> (You can [do it programmatically](https://github.com/fasterthanlime/feedback/issues/148), but since those dimensions dictate how large our canvas is, I chose something smaller than the theoretical maximum).
